typescript/test/client/client.test.ts

```diff
@@ -0,0 +1,193 @@
+    });
+
+    test.for(["failer", "raiser"])('%s run fails', async (agentName) => {
+      const client = createClient();
+
+      const run = await client.runSync(agentName, input);
+
+      expect(run.status).toBe('failed');
+      expect(run.error).toBeDefined();
+      expect(run.error?.code).toBe('invalid_input');
+    });
+
+    test.for(["awaiter", "slow_echo"])('%s run cancel works', async (agentName) => {
+      const client = createClient();
+
+      let run = await client.runAsync(agentName, input);
+      run = await client.runCancel(run.run_id);
+
+      expect(run.status).toBe('cancelling');
+
+      await delay(1000);
+
+      run = await client.runStatus(run.run_id);
+      expect(run.status).toBe('cancelled');
+    });
+
+    test('run cancel during a streaming works', async () => {
+      const client = createClient();
+
+      let lastEvent: Event | undefined;
+      for await (const event of client.runStream('slow_echo', input)) {
+        lastEvent = event;
+        if (event.type === 'run.created') {
+          const run = await client.runCancel(event.run.run_id);
+          expect(run.status).toBe('cancelling');
+        }
+      }
+      expect(lastEvent?.type).toBe('run.cancelled');
+    })
+
+    test('awaiter run is resumed sync', async () => {
+      const client = createClient();
+
+      let run = await client.runSync('awaiter', input);
+
+      expect(run.status).toBe('awaiting');
+      expect(run.await_request).toBeDefined();
+
+      run = await client.runResumeSync(run.run_id, awaitResume);
+
+      expect(run.status).toBe('completed');
+    });
+
+    test('awaiter run is resumed async', async () => {
+      const client = createClient();
+
+      let run = await client.runSync('awaiter', input);
+
+      expect(run.status).toBe('awaiting');
+      expect(run.await_request).toBeDefined();
+
+      run = await client.runResumeAsync(run.run_id, awaitResume);
+      expect(run.status).toBe('in-progress');
+    });
+
+    test('awaiter run is resumed stream', async () => {
+      const client = createClient();
+
+      let run = await client.runSync('awaiter', input);
+
+      expect(run.status).toBe('awaiting');
+      expect(run.await_request).toBeDefined();
+
+      const events: Event[] = [];
+      for await (const event of client.runResumeStream(run.run_id, awaitResume)) {
+        events.push(event);
+      }
+
+      expect(events.at(0)?.type).toBe('run.in-progress');
+      expect(events.at(-1)?.type).toBe('run.completed');
+    });
+
+    test('sessions work', async () => {
+      const client = createClient();
+      const session = client.withSession();
+
+      let run = await session.runSync('echo', input);
+      expect(run.output).toHaveLength(1);
+      run = await session.runSync('echo', input);
+      expect(run.output).toHaveLength(3);
+    })
+  });
+});
+
+
+function delay(ms: number): Promise<void> {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
```

Comment:
Tip (assuming tests run in node): https://nodejs.org/api/timers.html#timerspromisessettimeoutdelay-value-options